sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -0,0 +1,198 @@
+    return activity.getClass().getSimpleName();
+  }
+
+  private void startTracing(final @NonNull Activity activity) {
+    if (performanceEnabled && !isRunningTransaction(activity)) {
+      final ITransaction transaction =
+          hub.startTransaction(getActivityName(activity), "navigation");
+
+      // lets bind to the scope so other integrations can pick it up
+      hub.configureScope(
+          scope -> {
+            scope.withTransaction(
+                scopeTransaction -> {
+                  // we we'd not like to overwrite existent transactions bound to the Scope
+                  // manually.
+                  if (scopeTransaction == null) {
+                    scope.setTransaction(transaction);
+                  }
+                });
+          });
+
+      activitiesWithOngoingTransactions.put(activity, transaction);
+    }
+  }
+
+  private boolean isRunningTransaction(final @NonNull Activity activity) {
+    return activitiesWithOngoingTransactions.containsKey(activity);
+  }
+
+  private void stopTracing(final @NonNull Activity activity, final boolean shouldFinishTracing) {
+    if (performanceEnabled && shouldFinishTracing) {
+      final ITransaction transaction = activitiesWithOngoingTransactions.get(activity);
+      if (transaction != null) {
+        SpanStatus status = transaction.getStatus();
+        // status might be set by other integrations, let's not overwrite it
+        if (status == null) {
+          status = SpanStatus.OK;
+        }
+        //
+        transaction.finish(status);
+      }
+    }
+  }
+
+  @Override
+  public synchronized void onActivityPreCreated(
+      final @NonNull Activity activity, final @Nullable Bundle savedInstanceState) {
+    // if activity has global fields being init. and
+    // they are slow, this won't count the whole fields/ctor initialization time, but only
+    // when onCreate is actually called.
+    startTracing(activity);
+  }
+
+  @Override
+  public synchronized void onActivityCreated(
+      final @NonNull Activity activity, final @Nullable Bundle savedInstanceState) {
+    addBreadcrumb(activity, "created");
+  }
+
+  @Override
+  public synchronized void onActivityStarted(final @NonNull Activity activity) {
+    addBreadcrumb(activity, "started");
+  }
+
+  @Override
+  public synchronized void onActivityResumed(final @NonNull Activity activity) {
+    addBreadcrumb(activity, "resumed");
+  }
+
+  @Override
+  public synchronized void onActivityPostResumed(final @NonNull Activity activity) {
+    // this should be called only when onResume has been executed already, which means
+    // the UI is responsive at this moment.
+    stopTracing(activity, options.isEnableAutoActivityLifecycleTracingFinish());
+  }
+
+  @Override
+  public synchronized void onActivityPaused(final @NonNull Activity activity) {
+    addBreadcrumb(activity, "paused");
+  }
+
+  @Override
+  public synchronized void onActivityPostPaused(final @NonNull Activity activity) {
```

Comment:
`h`: I think this should be moved to onActivityStopped. OnPause is also called when you display a dialog. It just states the activity has no focus anymore, but it could be still visible.

Reply:
we can't, when you start a new activity, the new activity's onCreate is called before the current's activity onStop, it was like this before, onActivityPostPaused is the solution

Reply:
Oh, I didn't think about that. But is it fine to finish the transaction when you display a dialog?

Reply:
But what's the big deal calling this after the `onCreate` of the other activity is called? 

Another important point is for multi-window activities. The activity could be paused, but still visible, see https://developer.android.com/guide/topics/ui/multi-window#lifecycle. I know this is an edge case, but finishing the transaction on `onPause` feels a bit weird to me.

Reply:
@philipphofmann we have a single Hub and Scope, which takes only one transaction per time.
If I start a transaction before finishing the previous one, its not gonna set the new transaction to the Scope.
If we always overwrite the current transaction set to the Scope, its also a problem, as they wont be able to finish themselves as the transaction is not accessible anymore.

multi-window is indeed an issue, but I'd prefer to tackle this problem at a later problem, with many other problems that would come, probably.

Reply:
What about moving this to `onDestroy` and before starting a new transaction on `onCreate` we check if there is still an active transaction of another activity and just finish it?

Reply:
discussed along with https://github.com/getsentry/sentry-java/pull/1277#discussion_r585417497